### PR TITLE
[WIP] Add logging to find_package

### PR DESCRIPTION
**Pull request by iblancasa**
### Summary
CMake 3.15 added a cool new feature ([release notes](https://cmake.org/cmake/help/v3.15/release/3.15.html)):
> The message() command learned new types: NOTICE, VERBOSE, DEBUG and TRACE.

I am using this new feature in the FindPackage script for ConnextDDS.

### Details and comments
I implemented a function in order to wrapper the functionality. Why? This is something new in CMake 3.15 but I can make it work on previous releases (or something similar) using the macro and some logic.

The idea is to make easier to debug issues when we are looking for our libraries.

### CODE REVIEW

**Comment by iblancasa** [comment]
@jpovedano, @fgaranda, @lulivi: this feature is not finished but... feedback? Something interesting we can do with this?

File: resources/cmake/FindRTIConnextDDS.cmake [hunk 1]
```diff
@@ -264,6 +264,31 @@
 # - All the Darwin platforms (OS X 10.11-10.13)
 #
 
+#####################################################################
+# Logging Functions                                                  #
+#####################################################################
+function(connextdds_debug message)
+    message(DEBUG ${message})
```
Comment (jpovedano) [inline resources/cmake/FindRTIConnextDDS.cmake]: According to the documentation the DEBUG and VERBOSE levels were introduced in 3.15. Have we tried the behavior on previous versions? Were just undocumented levels?

**Comment by jpovedano** [comment]
I like it. Some ideas we may want to explore:

- Make the logging functions a macro, so we can make use of CMAKE_CURRENT_LIST_LINE for better debuggability if desired
- Add our own prefix to DEBUG/VERBOSE so messages are easier to fllter

File: resources/cmake/FindRTIConnextDDS.cmake [hunk 2]
```diff
@@ -264,6 +264,31 @@
...
+endfunction()
+
+function(connextdds_verbose message)
+    message(VERBOSE ${message})
```
Comment (fgaranda) [inline resources/cmake/FindRTIConnextDDS.cmake]: You could rename these functions to:
connextdds_log_verbose
connextdds_log_debug

or

connextdds_message_verbose,
connextdds_message_debug

to give a context about what they do.

**Comment by fgaranda** [comment]
Looks good! I have added a suggestion regarding the name, but other than that is pretty good.

File: resources/cmake/FindRTIConnextDDS.cmake (hunk 2 above)
Comment (lulivi) [reply resources/cmake/FindRTIConnextDDS.cmake]: I would go with `connextdds_log_xxxx` to match the function below (`connextdds_log_xml`)

**Pull request closed without merging**